elastic/kibana
# [maps] Add spatial filter from cluster

Fixes https://github.com/elastic/kibana/issues/128211

PR adds "Filter by cluster" action to cluster features. Clicking action will add [geo_grid filter](https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-geo-grid-query.html) to filter bar.

<img width="600" alt="Screen Shot 2022-06-06 at 2 26 55 PM" src="https://user-images.githubusercontent.com/373691/172242991-4640dee4-bc0c-49d4-8c17-8888317c7185.png">

<img width="600" alt="Screen Shot 2022-06-06 at 2 27 02 PM" src="https://user-images.githubusercontent.com/373691/172243019-0dd54368-1a62-485f-bc3c-64029c61f39c.png">

PR makes the following API changes
* remove ISource.getPreIndexedShape. Method leaked implementation details to API and is no longer needed.
* Add IVectorSource.getFeatureActions. This allows sources to customize tooltip actions and is used by this PR to add "Filter by cluster" for ESGeoGridSource and "Filter by geometry (pre-indexed shape)" for EsSearchSource.

To test
* Install sample web logs data set
* Create new map
*  Add "Clusters" layer
* Click cluster feature in map to lock tooltip. Click "Filter by cluster". Filter pill with geo_grid filter should be added to filter bar
